### PR TITLE
feat!: simplify configuration

### PR DESCRIPTION
**Pull request by beltram**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

`MlsConversationConfiguration` had some unused custom fields (`admins` & `key_rotation_span`). Since it wasn't used and probably won't be in a forseeable future those fields have been removed.  
Also, a configuration parameter was required when joining a group either by welcome or external commit. OpenMLS can infer the required missing fields from the configuration (`ciphersuite` & `external_senders`). So this configuration is not required anymore

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: crypto-ffi/bindings/js/CoreCrypto.ts [hunk 1]
```diff
@@ -58,20 +58,10 @@ export enum Ciphersuite {
...
-    admins?: Uint8Array[];
     /**
      * Conversation ciphersuite
      */
     ciphersuite?: Ciphersuite;
-    /**
-     * Duration in seconds after which we will automatically force a self_update commit
-     * Note: This isn't currently implemented
-     */
-    keyRotationSpan?: number;
```
Comment (OtaK) [inline crypto-ffi/bindings/js/CoreCrypto.ts:74]: We *will* need to implement this in the future, so we should keep it no matter what.

File: crypto-ffi/bindings/js/CoreCrypto.ts [hunk 2]
```diff
@@ -58,20 +58,10 @@ export enum Ciphersuite {
  * Configuration object for new conversations
  */
 export interface ConversationConfiguration {
-    /**
-     * List of client IDs with administrative permissions
-     * Note: This is currently unused
-     */
-    admins?: Uint8Array[];
```
Comment (OtaK) [inline crypto-ffi/bindings/js/CoreCrypto.ts:65]: This is fine to remove as our current authorization callbacks do the same thing basically

**Review by OtaK — CHANGES_REQUESTED** [review]

File: crypto-ffi/bindings/js/CoreCrypto.ts (hunk 1 above)
Comment (beltram) [reply crypto-ffi/bindings/js/CoreCrypto.ts:74]: I created another `CustomConfiguration` struct in addition to `ConversationConfiguration`. This struct will hold everything not natively handled by openmls ; curently `key_rotation_span` and `wire_format`.  
This struct will be used only where strictly required: joining from welcome or from external commit. That way, only the minimal amount of information is required on those methods without superfluous fields (i.e. ciphersuites).

File: crypto-ffi/bindings/js/CoreCrypto.ts [hunk 3]
```diff
@@ -595,21 +600,19 @@ export class CoreCrypto {
      * You will want to use {@link CoreCrypto.addClientsToConversation} afterwards to add clients to this conversation
      *
      * @param conversationId - The conversation ID; You can either make them random or let the backend attribute MLS group IDs
-     * @param configuration.admins - An array of client IDs that will have administrative permissions over the group
+     * @param configuration - configuration of the MLS group
      * @param configuration.ciphersuite - The {@link Ciphersuite} that is chosen to be the group's
-     * @param configuration.keyRotationSpan - The amount of time in milliseconds after which the MLS Keypackages will be rotated
      * @param configuration.externalSenders - Array of Client IDs that are qualified as external senders within the group
```
Comment (OtaK) [inline crypto-ffi/bindings/js/CoreCrypto.ts:605]: Should maybe add the `configuration.custom` field and redirecting to the `CustomConfiguration` interface there

**Review by OtaK — CHANGES_REQUESTED** [review]
+ remove the merge commits and use a rebase instead
+ some squashing needed

**Review by OtaK — APPROVED** [review]
Rebase + squash absolutely needed, all good otherwise

**Pull request merged**